components/Error.tsx

```diff
@@ -2,20 +2,27 @@ import React from 'react'
 import Layout from './Layout'
 import NavLink from './NavLink'
 import { Text } from './theme/Text'
-
+export enum StatusCode {
+  NOT_FOUND = '404',
+  INTERNAL_SERVER_ERROR = '500'
+}
 type ErrorProps = {
-  type: '404' | '500'
+  code: StatusCode
   message?: string
 }
-const Error: React.FC<ErrorProps> = ({ type, message }) => {
+const Error: React.FC<ErrorProps> = ({ code, message }) => {
+  const errorMessages: ReadonlyMap<StatusCode, string> = new Map([
```

Comment:
This hash table should be on the global scope. The table has constant values that will not change unless we manually update the file ourselves to add another error status code. As such, there is no need to generate another `errorMessages` table every time this component is used.

Reply:
Mapping from status codes is out of component scope and it's renamed to errorTitle.